app/packages/core/src/components/Actions/similar/Warning.tsx

```diff
@@ -1,29 +1,45 @@
-import React from "react";
 import { PopoutSectionTitle, useTheme } from "@fiftyone/components";
+import * as fos from "@fiftyone/state";
+import React from "react";
+import { useRecoilValue } from "recoil";
 import { SORT_BY_SIMILARITY } from "../../../utils/links";
 import { ActionOption } from "../Common";
 
 interface Props {
   hasSimilarityKeys: boolean;
   isImageSearch: boolean;
+  modal: boolean;
 }
 
 const Warning: React.FunctionComponent<Props> = (props) => {
   const theme = useTheme();
+  const hasSelectedSamples = useRecoilValue(fos.hasSelectedSamples);
+  const hasSelectedLabels = useRecoilValue(fos.hasSelectedLabels);
+  const hasSorting = Boolean(useRecoilValue(fos.similarityParameters));
+
+  const isImageSearch =
+    hasSelectedSamples ||
+    (props.isImageSearch && hasSorting) ||
+    (props.modal && hasSelectedLabels);
+
+  const warningText = isImageSearch
+    ? "Search by image similarity"
+    : "Sort by text similarity";
```

Comment:
can we move this to its own hook?

nit: write a unit test to abstract the state logic from UI logic

super nit: Also, can we change `warning` to `help` 

```
const { isImageSearch, modal } = props;
const {
  helpText
} = useSimilarityHelpText(isImageSearch, modal)
```

useSimilarityHelpText.ts
```
const useSimilarityHelpText(props) {

    const hasSelectedSamples = useRecoilValue(fos.hasSelectedSamples);
    const hasSelectedLabels = useRecoilValue(fos.hasSelectedLabels);
    const hasSorting = Boolean(useRecoilValue(fos.similarityParameters));

   const isImageSearch =
      hasSelectedSamples ||
      (props.isImageSearch && hasSorting) ||
      (props.modal && hasSelectedLabels);

  const text = isImageSearch
    ? "Search by image similarity"
    : "Sort by text similarity";

  return text
}
```

Reply:
I can write a hook that returns the similarity type, as it is used by both the icon and the helper text.